Perf triage for this week (#2100)
rust-lang/rustc-perf: triage/2025-05-04.md

# 2025-05-04 Triage Log

> A relatively noisy week due to addition of new benchmarks as part of our [2025
> benchmark update], and a number of large regressions in a rollup landing late
> in the week (and so not yet investigated).

Author: ```suggestion
A relatively noisy week due to addition of new benchmarks as part of our [2025
benchmark update], and a number of large regressions in a rollup landing late
in the week, with investigation [here](https://github.com/rust-lang/rust/pull/140652).
```

| Regressions ❌ <br /> (secondary)  | 0.4%  | [0.2%, 0.5%]   | 2     |
| Improvements ✅ <br /> (primary)   | -1.2% | [-1.2%, -1.2%] | 1     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -1.2% | [-1.2%, -1.2%] | 1     |

Regression triaged. Regressions look like bimodality/noise, not worth further follow up.

Rollup of 7 pull requests [#140633](https://github.com/rust-lang/rust/pull/140633) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=1bea580f364c65bd5f7380a1056e150df7b8a1a6&end=622ac043764d5d4ffff8de8cf86a1cc938a8a71b&stat=instructions:u)

| (instructions:u)                   | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 1.2%  | [0.2%, 13.4%]  | 77    |
| Regressions ❌ <br /> (secondary)  | 1.9%  | [0.2%, 11.0%]  | 155   |
| Improvements ✅ <br /> (primary)   | -1.5% | [-1.5%, -1.5%] | 1     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | 1.2%  | [-1.5%, 13.4%] | 78    |

> Still investigating the root cause as of writing (see comments on the PR).

Reviewer: We know now that the cause is https://github.com/rust-lang/rust/pull/139675 :)

> Still investigating the root cause as of writing (see comments on the PR).

Author: ```suggestion
Per investigation on the PR, the root cause is [#139675](https://github.com/rust-lang/rust/pull/139675). Per discussion [here](https://github.com/rust-lang/rust/pull/140652) we expect that this will be mostly resolved in our next LLVM bump, but is likely an acceptable regression in the meantime (primarily to very short compiles) for the feature in question.
```
